ambry-router/src/main/java/com.github.ambry.router/AdaptiveOperationTracker.java

```diff
@@ -77,15 +77,17 @@
   }
 
   @Override
-  public void onResponse(ReplicaId replicaId, boolean isSuccessFul) {
-    super.onResponse(replicaId, isSuccessFul);
+  public void onResponse(ReplicaId replicaId, boolean isSuccessFul, RouterErrorCode routerErrorCode) {
```

Comment:
how about just passing `boolean updateLatencyHistogram` instead of the error code? This will keep the tracker agnostic to the meaning of different error codes and will match what was done with the `isSuccessful` variable.

Reply:
Thanks for the suggestion. After reconsideration, it might be reasonable to keep router error code in method signature. Two humble thoughts: 1. I changed the `onResponse` signature in `OperationTracker` interface which is implemented by `SimpleOperationTracker` and `AdaptiveOperationTracker`. It seems `updateLatencyHistogram` doesn't apply to `SimpleOperationTracker`.  2. Passing router error code might give us some benefits in the future if we want to change behavior of both operation trackers based on the router error code.

Reply:
`isSuccessFul` is not able to provided enough information, so you introduce `RouterErrorCode`. 
But with both `isSuccessFul` and `RouterErrorCode`, it's a little redundant.

So I think there are two options,
1. Just pass in RouterErrorCode, internally distinguish `isSuccess` and `shoudUpdate`. 
2. Instead of passing in boolean, we can pass in a int(enum), different type do different things. 

Looks like 1 is better.

Reply:
ha, you are discussing in the thread below.

Reply:
took @cgtz 's suggestion and changed the method signature.